pkg/kubelet/kubelet_test.go

```diff
@@ -173,8 +173,12 @@ func newTestKubeletWithImageList(
 	kubelet.cadvisor = mockCadvisor
 
 	fakeMirrorClient := podtest.NewFakeMirrorClient()
-	fakeSecretManager := secret.NewFakeManager()
```

Comment:
Look at what it does. It doesn't work at all in any real case that will request a secret.
By using the real one, my tests now pass. Just ran into this in the optional configmap and secrets too.